diff -aur widelands-build15-src.pristine//src/editor/ui_menus/editor_main_menu_new_map.cc widelands-build15-src.new//src/editor/ui_menus/editor_main_menu_new_map.cc
--- widelands-build15-src.pristine//src/editor/ui_menus/editor_main_menu_new_map.cc	2010-05-22 14:13:40.641165379 +0200
+++ widelands-build15-src.new//src/editor/ui_menus/editor_main_menu_new_map.cc	2010-05-22 14:20:21.230312264 +0200
@@ -113,7 +113,7 @@
 		 posx, posy, width, height,
 		 g_gr->get_picture(PicMod_UI, "pics/but1.png"),
 		 &Main_Menu_New_Map::button_clicked, *this, 4,
-		 Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+                 Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 
 	posy += height + spacing + spacing + spacing;
 
@@ -142,8 +142,8 @@
 		if (m_currentworld == m_worlds.size())
 			m_currentworld = 0;
 		m_world->set_title
-			(Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name
-			 	());
+                    (Widelands::World(m_worlds[m_currentworld].c_str()).get_name
+                                    ());
 		break;
 	}
 
diff -aur widelands-build15-src.pristine//src/editor/ui_menus/editor_main_menu_random_map.cc widelands-build15-src.new//src/editor/ui_menus/editor_main_menu_random_map.cc
--- widelands-build15-src.pristine//src/editor/ui_menus/editor_main_menu_random_map.cc	2010-05-22 14:13:40.641165379 +0200
+++ widelands-build15-src.new//src/editor/ui_menus/editor_main_menu_random_map.cc	2010-05-22 14:29:19.827764161 +0200
@@ -272,7 +272,7 @@
 		 posx, posy, width, height,
 		 g_gr->get_picture(PicMod_UI, "pics/but1.png"),
 		 &Main_Menu_New_Random_Map::button_clicked, *this, 8,
-		 Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+                 Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 
 	posy += height + spacing + spacing + spacing;
 
@@ -343,7 +343,7 @@
 		if (m_currentworld == m_worlds.size())
 			m_currentworld = 0;
 		m_world->set_title
-			(Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+                    (Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 		break;
 	case 9:
 		break;
@@ -476,7 +476,7 @@
 			(strcmp(mapInfo.worldName.c_str(), m_worlds[m_currentworld].c_str()))
 			++m_currentworld;
 		m_world->set_title
-			(Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+                    (Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 
 		button_clicked(-1);  // Update other values in UI as well
 
Only in widelands-build15-src.new//src: widelands-gcc45.patch
